### PR TITLE
fix: pass state through navigate function

### PR DESCRIPTION
**Pull request by DenisIrkhin**
https://github.com/TanStack/router/issues/680 is raised. https://github.com/TanStack/router/pull/690 fixed the Link part but the `navigate` function is not passing `state` to `router.history.location` still.
This PR is supposed to fix it.

### CODE REVIEW

**Comment by DenisIrkhin** [comment]
The CI is failing. I appreciate guidance on how to make it green.

**Comment by DenisIrkhin** [comment]
I fixed the CI in https://github.com/TanStack/router/pull/720/commits/12b188f04c022ae9fee44af6a946cd941448119a
but not sure that it complies with the expected results. I reverted it as not related to the PR scope.

**Comment by tannerlinsley** [comment]
I think this was fixed with the route masking features. Confirm?

File: packages/router-core/src/router.ts [hunk 1]
```diff
@@ -1209,6 +1210,7 @@ export class Router<
       params,
       replace,
       resetScroll,
+      state,
```
Comment (DenisIrkhin) [inline packages/router-core/src/router.ts:1213]: I tried locally and on stackblitz
[forked beta v.188](https://stackblitz.com/edit/tanstack-router-bn5cgc?file=src%2Fmain.tsx)

`state` is not passing through `navigate` still...

Another point:
After the latest changes state object complies to `'true | NonNullableUpdater<LocationState> | undefined'`
image

that's not how it is on docs `// `state` is either an object of state or a function that supplies the previous state and allows you to return a new one. State is stored in the history API and can be useful for passing data between routes that you do not with to permanently store in URL search params.` 

I tried to use it as it was described on the docs - to store additional state and keep URL route simple without putting the state in search params.  I know it looks against one of the main concepts of TanStack router ;) , but as soon it's documented I tried to use it and open the PR.

**Comment by tannerlinsley** [comment]
There are some deeper issues here. I'll fix them right now.

**Comment by tannerlinsley** [comment]
Pull the latest and let me know how things are.

**Comment by DenisIrkhin** [comment]
I see your changes.
`state` is passing to `location` on the beta.190
Thank You.

I'm closing the PR.

**Pull request closed without merging**